src/audio/src/src.c

```diff
@@ -987,6 +987,13 @@ static int src_init(struct processing_module *mod)
 
 	comp_dbg(dev, "src_init()");
 
+	if ((IS_ENABLED(CONFIG_IPC_MAJOR_3) && dev->ipc_config.type != SOF_COMP_SRC) ||
+	    !cfg->init_data || cfg->size != sizeof(cd->ipc_config)) {
+		comp_err(dev, "src_init(): Missing or bad size (%u) init data",
```

Comment:
this is complex condition, and the err message can't cover so much cases.
if for bugfix, we can simply add:
#if CONFIG_IPC_MAJOR_3
if (!cfg->init_data) {
     comp_err(dev, "src_init(): Missing or bad size (%u) init data",cfg->size);
    return -EINVAL;
}
#endif

for handle other cases, how about use separate "if"?

Reply:
The IPC config type check is now separate, and inside #if IPC3. Functionally this should be equivalent to the previous version.